proposed/2022/dotnet-nuget-why-proposal.md

```diff
@@ -0,0 +1,124 @@
+
+```
+dotnet nuget why packageA -f net6.0
+
+Project 'projectNameA' has the following dependency graph for 'packageA'
+   [net6.0]: Microsoft.ML (1.0.0) -> Microsoft.ML.Util (1.0.0) -> packageA (1.0.0)
+
+Project 'projectNameB' has the following dependency graph for 'packageA'
+   [net6.0]: Microsoft.ML (1.1.0) -> Microsoft.ML.Util (1.1.0) -> packageA (1.1.0)
+```
+
+## Drawbacks 
+
+The contents of the `project.assets.json` will be helpful in understanding the dependency graph of a given package but there are a few limitations:
+
+-  There may be some packages in the file that are not part of the final graph because the NuGet restore operation downloads other packages during dependency resolution. The absence of a package id and version in the `project.assets.json` file indicates that a package was downloaded during dependency resolution but is not part of the final dependency graph.
+- Packages acquired through `PackageDownload` are not tied to the project in any way beyond acquisition. These packages are not recorded in the final graph that `project.assets.json` maintains, however they are recorded under project -> frameworks -> {frameworkName} -> downloadDependencies.
+
+## Rationale and Alternatives
+
+We could have considered a minor tweak to the existing experience of “dotnet list package --include-transitive” to provide the user with a sense of a package's dependencies. This could have been a new column next to “Resolved” called “Transitively Referenced” that had a list of the top-level packages that requested the dependency. The corresponding tracking issue can be found here: https://github.com/NuGet/Home/issues/11625. Unfortunately, this solution doesn't provide an easy way to identify the dependency graph `(starting from project -> top-level package -> transitive-dependency (1) -> ..-> transitive dependency (n))` for a particular package.
+
+![](../../meta/resources/TransitiveDependencies/TransitiveDotNetCLI.png)
+
+## Prior Art
+
+Within Visual Studio, there will be a new panel within the Visual Studio Package Manager UI named “Transitive Packages” in which all transitive packages will be displayed to the user. For the sake of not confusing the user, there will be an additional header titled “Top-level Packages”.
+
+When a user highlights a transitive package, they will see a pop-up that displays how the transitive dependency originated & what top-level package(s) are bringing it in.
+
+![](../../meta/resources/TransitiveDependencies/TransitiveVSPMUI.png)
+
+
+## Unresolved Questions
+
+Projects that use `packages.config` format are out of scope for this effort because the XML file contains a flat list of both direct and transitive package dependencies. Customers cannot visualize the dependency graph from the file itself but it would be easy to update the package version of a transitive depedency if there are any security vulnerabilities in the current version.
```

Comment:
The .NET CLI (and therefore `dotnet restore`) doesn't support packages.config, so I don't think we should even consider supporting it in other `dotnet nuget` commands. I'd consider it completely out of scope, rather than an unresolved question.

Reply:
Sure, that makes sense, I'll remove it from the spec.